<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>珠峰培训</title>
</head>
<body>
    <div id="div1">珠峰培训</div>
</body>
</html>
<script src="js/ajax.js"></script>
<script>

    var p = new Promise(function (resolve,reject) {
        ajax({
            type:'get',
            url:'./1.json',
            // success:resolve,// success && success(data)
            success:function(data){
                // data[19] = 1233;
                resolve(data);
            },
            error:reject
        });
    });
    let p2 = p.then((data)=>{
        console.log(data);
        return new Promise(function (resolve,reject) {
            setTimeout(function () {
                resolve('success')
            },1000);
            reject(data);//走这 执行.then的第二个函数
        })
    },()=>{});
    // 可以理解成 p2 = new Promise(function (resolve,reject) {
    //                     setTimeout(function () {
    //                         resolve('success')
    //                     },1000);
    //                     reject(data);
    //                 })
    p2.then(data=>{
        console.log(data);
    },res=>{
        setTimeout(function () {
            console.log(res,j);
        },2000);
        return 123
        //走这  计时器没执行就会直接 执行下面第一个函数 因为计时器成功执行 就算成功
        //so  会先输出下面的data没传参 所以是undefined  然后报错 没定义q
    }).then((data)=>{
        console.log(data);
    },()=>{})

</script>  